.github/workflows/pr-e2e-testing.yaml

```
@@ -64,3 +64,11 @@ jobs:
       - name: Run E2E Tests
         if: ${{ steps.filter.outputs.notdocs == 'true' }}
         run:  make run-e2e-test VERBOSE=true # use DEBUG_VM_CONSOLE=1 to see the VM console output
+
+      - name: Collect and Upload Logs
+        if: always() && ${{ steps.filter.outputs.notdocs == 'true' }}
```

actionlint (1.7.4)

if: condition "always() && ${{ steps.filter.outputs.notdocs == 'true' }}" is always evaluated to true because extra characters are around ${{ }}

(if-cond)